tests/pos/i15029.more.scala

```diff
@@ -0,0 +1,25 @@
+// scalac: -Werror
+
+// Like tests/pos/i15029.scala, but with a more complicated prefix.
+
+sealed trait Schema[A]
+
+sealed class Outer:
+  sealed trait RecordInstances[B]:
+    case class Field() extends Schema[B]
+    case object Thing extends Schema[B]
+
+object O1 extends Outer
+object O2 extends Outer
+
+object X extends O1.RecordInstances[Int]
+object Y extends O2.RecordInstances[String]
+
+// Match not exhaustive error! (with fatal warnings :P)
+class Test:
+  def handle[T](schema: Schema[T]) =
+    schema match // was: match may not be exhaustive
+      case X.Field() =>
+      case X.Thing   =>
+      case Y.Field() =>
+      case Y.Thing   =>
```

Comment:
In the example mentioned above, the code is as follows:

```Scala
  def handle(schema: Schema[String]) =
    schema match // was: match may not be exhaustive
      case Y.Field() =>
      case Y.Thing   =>
```

Reply:
So, in that case it still does the right thing, though the `showType` logic isn't handling it very well.

Reply:
Sorry, I was not clear. If I run the following example with this PR:

```Scala
// scalac: -Werror
sealed trait Schema[A]

sealed class Outer:
  sealed trait RecordInstances[B]:
    case class Field() extends Schema[B]
    case object Thing extends Schema[B]

object O1 extends Outer
object O2 extends Outer

object X extends O1.RecordInstances[Int]
object Y extends O2.RecordInstances[String]

// Match not exhaustive error! (with fatal warnings :P)
class Test:
  def handle(schema: Schema[String]) =
    schema match // was: match may not be exhaustive
      case Y.Field() =>
      case Y.Thing   =>

```

I get the following warning:

``` Scala
-- [E029] Pattern Match Exhaustivity Warning: tests/pos/i15029.more.scala:18:4 -
18 |    schema match // was: match may not be exhaustive
   |    ^^^^^^
   |    match may not be exhaustive.
   |
   |    It would fail on pattern case: X.Field(), Thing
   |
   | longer explanation available when compiling with `-explain`
```

I expect no warnings in this case.

Reply:
Ah, my mistake, didn't see the `Schema[String]` either time.

Reply:
Excellent, found a simplier way to do it, which makes your more complicated version work! ❤️